### PR TITLE
fog conformance tests: add a happy path 2nd server + retire test

### PR DESCRIPTION
**Pull request by eranrund**
Soundtrack of this PR: https://soundcloud.com/janoma/janoma-live-sisyphos-the-bronny-bounce

### Motivation

Currently conformance tests only utilize a single ingest server. We want to make the tests more elaborate and have them cover multiple scenarios that involve numerous ingest servers. This PR is the first step in that direction.

### In this PR
* Change fog ingest client utility to output JSON ingest summary data and consume it in the Python code
* Uprev `mobilecoin`
* Add a test that spawns a 2nd ingest server, instructs the 1st one to retire, send transactions using the second server's ingest key and verify they are found using the test client.

### Future Work
* More elaborate tests.

### CODE REVIEW

File: tools/fog-local-network/local_fog.py [hunk 1]
```diff
@@ -80,7 +81,7 @@ def start(self):
             f'--peer-listen-uri=insecure-igp://localhost:{self.peer_port}/',
             f'--ias-api-key={IAS_API_KEY}',
             f'--ias-spid={IAS_SPID}',
-            f'--local-node-id localhost:{self.client_port}',
```
Comment (garbageslam) [inline tools/fog-local-network/local_fog.py:83]: thank you!

File: tools/fog-local-network/fog_conformance_tests.py [hunk 2]
```diff
@@ -564,23 +557,16 @@ def run(self):
...
-        self.follow_up_balance_check(wallet0_from4a, {5: 10}, [5])
-        self.follow_up_balance_check(wallet1_from4a, {5: 6}, [5])
-        self.follow_up_balance_check(wallet2_from4a, {5: 0}, [5])
-        self.follow_up_balance_check(wallet3_from4a, {5: 0}, [5])
-        self.follow_up_balance_check(wallet4_from4a, {5: 0}, [5])
+        for (wallet0, wallet1, wallet2, wallet3, wallet4) in (
+            (wallet0_from1, wallet1_from1, wallet2_from1, wallet3_from1, wallet4_from1),
+            (wallet0_from4, wallet1_from4, wallet2_from4, wallet3_from4, wallet4_from4),
+            (wallet0_from4a, wallet1_from4a, wallet2_from4a, wallet3_from4a, wallet4_from4a),
+        ):
```
Comment (garbageslam) [inline tools/fog-local-network/fog_conformance_tests.py:564]: interesting -- i did not know this works in python

one thing is, when the test fails, you won't get a line number showing which one it is, but it there is a "name" for the wallet like wallet3_from1 that gets reported when balance check reports an error

Comment (eranrund) [reply tools/fog-local-network/fog_conformance_tests.py:564]: I'll look into ensuring the name associated with the wallets is printed before each check.

**Review by garbageslam — APPROVED** [review]
this looks good to me, thank you!

File: tools/fog-local-network/fog_conformance_tests.py [hunk 3]
```diff
@@ -784,77 +720,130 @@ def run(self):
...
+        # Both ingests should currently be active
+        status = self.fog_ingest.get_status()
+        assert status["mode"] == "Active", status
+
+        status = self.fog_ingest2.get_status()
+        assert status["mode"] == "Active", status
+
+        # Add block 9 to ingest and ledger. This should cause ingest1 to become Idle.
+        # Give 1 to everyone
+        # Take 2 from ???
```
Comment (mfaulk) [inline tools/fog-local-network/fog_conformance_tests.py:804]: ???

File: fog/ingest/client/src/main.rs [hunk 4]
```diff
@@ -54,12 +56,14 @@ fn main() -> ClientResult<()> {
 fn get_status(logger: &Logger, ingest_client: &FogIngestGrpcClient) -> ClientResult<()> {
     let status = ingest_client.get_status().expect("rpc failed");
     log::info!(logger, "Status: {:?}", status);
+    println!("{}", ingest_summary_to_json(&status));
```
Comment (sugargoat) [inline fog/ingest/client/src/main.rs:59]: What's the reason for the printlns? Is this read by the conformance test?

Comment (eranrund) [reply fog/ingest/client/src/main.rs:59]: Yup, this allows scripts to pick up the output and parse it.

Comment (sugargoat) [reply fog/ingest/client/src/main.rs:59]: I see - neat :)

**Review by sugargoat — APPROVED** [review]

**Pull request merged**